data-prepper-api/src/main/java/org/opensearch/dataprepper/model/failures/DlqObject.java

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.model.failures;
+
+import org.apache.commons.lang3.StringUtils;
+
+import java.time.Instant;
+import java.time.ZoneId;
+import java.time.format.DateTimeFormatter;
+import java.util.Objects;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkNotNull;
+
+/**
+ * A model representing DLQ objects in Data Prepper
+ *
+ * @since 2.2
+ */
+public class DlqObject {
+
+    private static final String ISO8601_FORMAT_STRING = "yyyy-MM-dd'T'HH:mm:ss.SSS'Z'";
+    private static final DateTimeFormatter FORMATTER = DateTimeFormatter.ofPattern(ISO8601_FORMAT_STRING)
+            .withZone(ZoneId.systemDefault());;
+
+    private static final String VERSION = "1";
```

Comment:
why does this class need specific versioning?

Reply:
It is a version on the DLQ object. If we change the overall structure in a breaking way, we can increment this value and users would know to parse the data according to the new version.

Reply:
I don't think it's a good idea to have separate versioning for internal components. If you make a breaking version change here, change the Data Prepper version.

Reply:
Providing a version was part of the original issue #2298. This model may change in the future and supporting a version mechanism will allow us to indicate this to the consumers.

Reply:
Versioning inside something that already has versioning is an anti-pattern in software. Would you build a separate versioning system inside of a Brazil package that already has versioning or would you use the Brazil versioning system to denote breaking changes?

Reply:
@asuresh8 ,

I take it you are concerned that this could be confusing because Data Prepper 4.x could be using a DLQ version 2. 

I would expect that this version does not change very frequently. And it is connected to the data format rather than the Data Prepper runtime. Would it be clearer if this were title `dlq_version`?

And do you have any suggestions for how to know if the format actually breaks? For example, there may be multiple Data Prepper major version changes before the DLQ version were to break?

Reply:
You should tie breaking changes in any class to the major version. Ex. If Data Prepper latest version is 4.x and you make a breaking change in this class, then increase the major version of all of Data Prepper to 5.x. Ideally, you want to group all breaking changes in one major version bump. So you may stage 5.x in a separate branch and keep adding 4.x -> 5.x breaking changes there. Then at some point, you release 5.x.